docs/developing-fuzzbench/adding_a_new_benchmark.md

```diff
@@ -71,86 +61,147 @@ Add the files in the benchmark directory to git (and then commit them):
 
-This file builds the fuzz target for your benchmark, taking into account
-the compiler options - `CC`, `CXX`, `CFLAGS`, `CXXFLAGS` and the fuzzer library
+This file builds the fuzz target for your benchmark. It should use the
+environment variables `CC`, `CXX`, `CFLAGS`, `CXXFLAGS` `$OUT` and the fuzzer
+library.
 `FUZZER_LIB` ([explanation]({{ site.baseurl }}/getting-started/adding-a-new-fuzzer/#what-is-fuzzer_lib))
-as input.
+for this. These environment variables will be defined when `build.sh` runs
+By linking your fuzz target with the `FUZZER_LIB` and the project under test,
+you will produce a binary that can be fuzzed by the FuzzBench fuzzers. This is
+called the fuzz target binary. `FUZZER_LIB` is specific to each fuzzer, its
+primary purpose is taking data from the fuzzer and passing it to your
+benchmark's `LLVMFuzzerTestOneInput` function.
 
 Once the build is finished, copy the fuzz target binary, any library
-dependencies and the seeds directory into the output directory (`$OUT`).
+dependencies (and optionally the seeds directory and the dictionary)
+into the output directory (`$OUT`).
 **NOTE**: Only build artifacts added in `$OUT` directory are available when
 running the fuzzer. You should not have any dependencies outside of `$OUT`.
 
-
 ```bash
 #!/bin/bash -ex
-. $(dirname $0)/../common.sh
 
-build_project() {
-  rm -rf BUILD
-  cp -rf SRC BUILD
-  (cd BUILD && ./autogen.sh && ./configure && make -j $JOBS)
-}
-get_git_revision <path-to-git-repo> <git-hash> SRC
-build_project
+rm -rf BUILD
+cp -rf $SRC/<project-source-code> BUILD
+cd BUILD && ./autogen.sh && ./configure && make -j
+
 
 # Build fuzz target in $OUT directory.
+export FUZZ_TARGET_NAME=fuzz_target
 $CXX $CXXFLAGS ${SCRIPT_DIR}/fuzz_target.cc \
     -I BUILD/path/to/include/dir BUILD/path/to/project-lib.a \
     $FUZZER_LIB \
-    -o $OUT/fuzz_target
+    -o $OUT/$FUZZ_TARGET_NAME
+
+# Optional: Copy seeds directory to $OUT directory.
+cp -r seeds $OUT/
 
-# Optional. Copy seeds directory to $OUT directory.
-cp -r $SCRIPT_DIR/seeds $OUT/
+# Optional: Copy dictionary to $OUT directory.
+cp $FUZZ_TARGET.dict $OUT/
 ```
 
 Example: [libxml2](https://github.com/google/fuzzbench/blob/master/benchmarks/libxml2-v2.9.2/build.sh).
 
-### seeds directory (optional)
+#### `seeds` directory (optional)
 
 This directory should contain a set of test input files for the fuzz target that
 provide good code coverage to start from. This should be copied to `$OUT/seeds`
 
 Example: [libpng-1.2.56](https://github.com/google/fuzzbench/blob/master/benchmarks/libpng-1.2.56/seeds).
 
+
+#### Dictionary file (optional)
+
+In the `$OUT` directory, you can define a file that will be used by fuzzers as a
+[dictionary](https://llvm.org/docs/LibFuzzer.html#id31). This file have the same
+name as the fuzz target binary followed by a `.dict` file extension. For example
+if your fuzz target binary is `$OUT/fuzz-target` the dictionary should be
+`$OUT/fuzz-target.dict`.
+
+### Dockerfile
```

Comment:
Shouldnt Dockerfile come become build.sh section ?

Reply:
Did you mean before it?
I actually think it makes sense to do afterwards since the build.sh is the important part, the Dockerfile is almost unnecessary and is simply to setup things for build.sh to run. At least the way I have it written now where I put the dictionary and seeds explanation in build.sh section it makes sense to put the build.sh section first.